src/io.jl

```diff
@@ -419,7 +421,7 @@ end
 function _writecallback(png_ptr::png_structp, data::png_bytep, length::png_size_t)::Csize_t
     a = png_get_io_ptr(png_ptr)
     io = unsafe_pointer_to_objref(a)
-    unsafe_write(io, data, length)
+    unsafe_write(io[], data, length)
```

Comment:
I believe you can actually fold these together, and slightly simplify the resulting operations:
```suggestion
    io = unsafe_load(Ptr{Any}(a))
    unsafe_write(io, data, length)
```

Reply:
Great, yeah that works